src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_NUMBER = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size >= array.length) {
+            int newCapacity = (int) (array.length * GROWTH_NUMBER);
+            if (newCapacity <= size) {
+                newCapacity = size + 1;
+            }
+            array = Arrays.copyOf(array, newCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Such index can not be add" + index);
```

Comment:
Add whitespace after word:
`throw new ArrayListIndexOutOfBoundsException("Such index can not be add: " + index);`